WebDevStudios/nextjs-wordpress-starter
# Feature/45 input error component

Closes #86

### Link

https://nextjs-wordpress-starter-9s7u9g468.vercel.app/form-demo

### Description

This PR contains updates around cleaning up the GravityForm integration.

- Doc block updates
- Add `InputerError` component to provide a wrapper for displaying errors.
- Add `isRequired` prop to all GravityForm fields
- Add minor styling to better indicate required fields and errors
- Refactor validation schema and form defaults into top level component
- Abstract functions into external files
- Add checkbox helper for setting up defaults
- Add select helper for setting up defaults
- Update `CheckboxGroup` and `Checkbox` inputs to use a value prop.
- Setup a merged data object for the GravityForm `Checkbox` field.
- Remove extra wrapper on fields. Note: GravityForms Fields will are streamlined for just mapping props from the field to the Input needed to render the field.
- Fixed Icon doc block comment

### Screenshot
<img width="688" alt="Screen Shot 2021-01-20 at 2 31 28 PM" src="https://user-images.githubusercontent.com/25035900/105225182-3329d400-5b2c-11eb-8062-5c801bb47f76.png">

**Checkbox default values**
While this feature may be enhanced later, default values based are working at this time.

<img width="508" alt="Screen Shot 2021-01-20 at 2 05 16 PM" src="https://user-images.githubusercontent.com/25035900/105225196-3755f180-5b2c-11eb-995d-2d3838d845af.png">

<img width="522" alt="Screen Shot 2021-01-20 at 2 05 10 PM" src="https://user-images.githubusercontent.com/25035900/105225194-3755f180-5b2c-11eb-9787-7f505014a3f7.png">

**Select default values**
While this feature may be enhanced later, default values based are working at this time.

<img width="525" alt="Screen Shot 2021-01-20 at 2 32 55 PM" src="https://user-images.githubusercontent.com/25035900/105225705-f3172100-5b2c-11eb-8c6b-9d818348aa4c.png">

<img width="431" alt="Screen Shot 2021-01-20 at 2 35 50 PM" src="https://user-images.githubusercontent.com/25035900/105225708-f4484e00-5b2c-11eb-9bb8-91e3aecd542d.png">

### Verification

How will a stakeholder test this?

1. Checkout update fields such as the select & checkbox.
2. Verify defaults work as expected.
